packages/playwright-core/src/server/browser.ts

```diff
@@ -103,10 +103,10 @@ export abstract class Browser extends SdkObject {
   _videoStarted(context: BrowserContext, videoId: string, path: string, pageOrError: Promise<Page | Error>) {
     const artifact = new Artifact(context, path);
     this._idToVideo.set(videoId, { context, artifact });
-    context.emit(BrowserContext.Events.VideoStarted, artifact);
```

Comment:
We probably have a test that ensures all videos are reported, even for those popups that were closed very fast.

Reply:
I don't think we do! But this makes me think I should try skipping video recording there!